File: scripts/capture-difc-ppt-screenshots.mjs
/**
 * Capture all DIFC Executive Intelligence screens (except Settings) for PPT export.
 * Requires the standalone HTML served over HTTP (see generate-difc-ppt.mjs).
 */
import { createRequire } from 'node:module';
import { mkdir, cp } from 'node:fs/promises';
import path from 'node:path';
import { fileURLToPath } from 'node:url';

const require = createRequire(import.meta.url);
const { chromium } = require(
  '/Users/arunkumarg/Desktop/AI Projects/AI Catalog App - Naar/naar-web-app/node_modules/playwright',
);

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const ROOT = path.resolve(__dirname, '..');
const OUT = path.join(ROOT, 'docs', 'deck-screenshots', 'difc-executive-intelligence');
const DOWNLOADS = path.join(
  process.env.HOME || '',
  'Downloads',
  'difc-executive-intelligence-ppt',
);
const BASE =
  process.env.DIFC_SCREENSHOT_BASE ||
  'http://127.0.0.1:8765/DIFC%20Executive%20Intelligence%20%28standalone%29.html';

const SHOTS = [
  { name: '01-executive-home', screen: 'home', label: 'Executive Home', wait: 2500 },
  { name: '02-ask-personal-ai-agent', screen: 'agent', label: 'Ask Personal AI Agent', wait: 2000 },
  { name: '03-performance', screen: 'performance', label: 'Performance', wait: 2000 },
  { name: '04-market-intelligence', screen: 'market', label: 'Market Intelligence', wait: 2500 },
  { name: '05-regulatory', screen: 'regulatory', label: 'Regulatory', wait: 2000 },
  { name: '06-knowledge-base', screen: 'knowledge', label: 'Knowledge Base', wait: 2000 },
  { name: '07-briefings', screen: 'briefings', label: 'Briefings', wait: 2000 },
];

const LAUNCHERS = [
  () =>
    chromium.launch({
      headless: true,
      executablePath:
        '/Users/arunkumarg/Library/Caches/ms-playwright/chromium_headless_shell-1223/chrome-headless-shell-mac-arm64/chrome-headless-shell',
    }),
  () =>
    chromium.launch({
      headless: true,
      executablePath:
        '/Users/arunkumarg/Library/Caches/ms-playwright/chromium-1223/chrome-mac-arm64/Google Chrome for Testing.app/Contents/MacOS/Google Chrome for Testing',
    }),
  () => chromium.launch({ headless: false, channel: 'chrome' }),
  () => chromium.launch({ headless: true, channel: 'chrome' }),
];

async function launchBrowser() {
  for (const tryLaunch of LAUNCHERS) {
    try {
      return await tryLaunch();
    } catch {
      /* next */
    }
  }
  throw new Error('Could not launch a browser for screenshots');
}

async function main() {
  const ping = await fetch(BASE).catch(() => null);
  if (!ping?.ok) {
    throw new Error(
      `DIFC app not reachable at ${BASE}. Start: python3 -m http.server 8765 --bind 127.0.0.1 (from Downloads)`,
    );
  }

  await mkdir(OUT, { recursive: true });
  await mkdir(DOWNLOADS, { recursive: true });

  const browser = await launchBrowser();
  const context = await browser.newContext({
    viewport: { width: 1440, height: 900 },
    deviceScaleFactor: 2,
  });

  const page = await context.newPage();

  for (const shot of SHOTS) {
    console.log(`→ ${shot.label}`);
    await page.goto(BASE, { waitUntil: 'networkidle', timeout: 90000 });
    await page.evaluate((screenId) => {
      localStorage.setItem('difc.screen', screenId);
    }, shot.screen);
    await page.reload({ waitUntil: 'networkidle', timeout: 90000 });
    await page.waitForTimeout(shot.wait);
    await page.waitForSelector('.app', { timeout: 30000 });

    const file = path.join(OUT, `${shot.name}.png`);
    await page.screenshot({ path: file, fullPage: false });
    await cp(file, path.join(DOWNLOADS, `${shot.name}.png`));
    console.log(`  ${file}`);
  }

  await browser.close();
  console.log(`\n${SHOTS.length} screenshots →\n  ${OUT}\n  ${DOWNLOADS}`);
}

main().catch((err) => {
  console.error(err.message || err);
  process.exit(1);
});

File: scripts/capture-difc-ppt-screenshots.py
#!/usr/bin/env python3
"""Capture all DIFC Executive Intelligence screens (except Settings) for PPT export."""
from __future__ import annotations

import os
import shutil
import sys
import time
from pathlib import Path

from playwright.sync_api import sync_playwright

ROOT = Path(__file__).resolve().parents[1]
OUT = ROOT / "docs" / "deck-screenshots" / "difc-executive-intelligence"
DOWNLOADS = Path.home() / "Downloads" / "difc-executive-intelligence-ppt"
BASE = os.environ.get(
    "DIFC_SCREENSHOT_BASE",
    "http://127.0.0.1:8765/DIFC%20Executive%20Intelligence%20%28standalone%29.html",
)

SHOTS = [
    ("01-executive-home", "home", "Executive Home", 2500),
    ("02-ask-personal-ai-agent", "agent", "Ask Personal AI Agent", 2000),
    ("03-performance", "performance", "Performance", 2000),
    ("04-market-intelligence", "market", "Market Intelligence", 2500),
    ("05-regulatory", "regulatory", "Regulatory", 2000),
    ("06-knowledge-base", "knowledge", "Knowledge Base", 2000),
    ("07-briefings", "briefings", "Briefings", 2000),
]


def main() -> int:
    OUT.mkdir(parents=True, exist_ok=True)
    DOWNLOADS.mkdir(parents=True, exist_ok=True)

    with sync_playwright() as p:
        browser = None
        last_err = None
        for launcher in (
            lambda: p.chromium.launch(headless=True),
            lambda: p.chromium.launch(headless=True, channel="chrome"),
            lambda: p.chromium.launch(headless=False, channel="chrome"),
        ):
            try:
                browser = launcher()
                break
            except Exception as exc:
                last_err = exc
        if browser is None:
            print(f"ERROR: Could not launch a browser for screenshots: {last_err}", file=sys.stderr)
            return 1

        context = browser.new_context(
            viewport={"width": 1440, "height": 900},
            device_scale_factor=2,
            color_scheme="light",
        )
        page = context.new_page()

        for name, screen, label, wait_ms in SHOTS:
            print(f"→ {label}")
            page.goto(BASE, wait_until="networkidle", timeout=90000)
            page.evaluate("(id) => localStorage.setItem('difc.screen', id)", screen)
            page.reload(wait_until="networkidle", timeout=90000)
            page.wait_for_selector(".app", timeout=30000)
            time.sleep(wait_ms / 1000)
            out_file = OUT / f"{name}.png"
            page.screenshot(path=str(out_file), full_page=False, type="png")
            shutil.copy2(out_file, DOWNLOADS / f"{name}.png")
            print(f"  {out_file}")

        browser.close()

    print(f"\n{len(SHOTS)} screenshots →\n  {OUT}\n  {DOWNLOADS}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
